dev/tasks/r/github.packages.yml

```diff
@@ -137,36 +137,44 @@ jobs:
       matrix:
         platform:
           - windows-latest
-          # This is newer than what CRAN builds on, but Travis is no longer an option for us, so...
-          - macos-10.15
+          - macos-10.13 # self-hosted
           # - devops-managed # No M1 until the runner application runs native
         r_version:
           - { rtools: 40, r: "4.1" }
           - { rtools: 42, r: "4.2" }
     steps:
       - uses: r-lib/actions/setup-r@v2
+        if: matrix.platform != 'macos-10.13'
         with:
           r-version: {{ '${{ matrix.r_version.r }}' }}
           rtools-version: {{ '${{ matrix.r_version.rtools }}' }}
           Ncpus: 2
+      - name: Setup R Self-Hosted
+        if: matrix.platform == 'macos-10.13'
+        run: |
+          # rig is a system utility that allows for switching
+          # between pre-installed R version on the self-hosted runners
+          rig default {{ '${{ matrix.r_version.r }}' }}
+          rig system setup-user-lib
+          rig system add-pak 
       {{ macros.github_setup_local_r_repo(false, true)|indent }}
+      - name: Install dependencies
+        shell: bash
+        run: |
+          tar -xzf repo/src/contrib/arrow_*.tar.gz arrow/DESCRIPTION

```

Comment:
I'm curious, how is this line supposed to install dependencies?

Reply:
You are right the name isn't ideal, the next step actually installs the dependencies but this step is needed for it, so maybe "prepare dependency installation" or something?

Reply:
Sounds good.